From 07af283f6ec0362620691a54a5a6549ba9478437 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Edstr=C3=B6m?= <meedstrom@runbox.eu>
Date: Sat, 17 May 2025 12:25:29 +0200
Subject: [PATCH] Ignore buffer-env

---
 el-job.el | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/el-job.el b/el-job.el
index 8d9f191..932ec3e 100644
--- a/el-job.el
+++ b/el-job.el
@@ -518,6 +518,10 @@ see `el-job-launch'."
            "--batch"
            "--load" (el-job--ensure-compiled-lib 'el-job-child)
            "--eval" (format "(el-job-child--work #'%S)" funcall-per-input)))
+         ;; Ignore buffer-env
+         ;; https://github.com/meedstrom/org-node/issues/98
+         (process-environment (default-value 'process-environment))
+         (exec-path (default-value 'exec-path))
          ;; Ensure the working directory is not remote.
          ;; https://github.com/meedstrom/org-node/issues/46
          (default-directory invocation-directory)
