diff --git i/bus/Makefile.am w/bus/Makefile.am
index bd3481b9..11df44ad 100644
--- i/bus/Makefile.am
+++ w/bus/Makefile.am
@@ -128,6 +128,14 @@ TESTS_ENVIRONMENT = \
     DISABLE_DAEMONIZE_IN_TESTS="$(DISABLE_DAEMONIZE_IN_TESTS)" \
     $(NULL)
 
+if ENABLE_GTK4
+TESTS_ENVIRONMENT += TEST_GTK_VERSION=4
+else
+if ENABLE_GTK3
+TESTS_ENVIRONMENT += TEST_GTK_VERSION=3
+endif
+endif
+
 LOG_COMPILER = $(top_srcdir)/src/tests/runtest
 
 noinst_PROGRAMS = $(TESTS)
